soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -381,4 +381,28 @@ void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 	z_xt_ints_on(core_desc[cpu].intenable);
 }
 
+#ifdef CONFIG_ARCH_CPU_IDLE_CUSTOM
+
+#include <pm_data.h>
+
+void arch_cpu_idle(void)
```

Comment:
Instead of having it, couldn't you use a new power state  ? A new runtime-idle with a different substate  and residency time ? Then you just need to add it in the existent `pm_state_set`

Reply:
At first, I thought about it and considered it a stylistically nicer solution. However, in this case, all I have to do is change the clock and then transition to the WAITI state. I'm concerned that the entire logic provided by the execution of code from `pm_system_suspend` may result in us losing any potential benefits that this entire operation is supposed to deliver.